osbs/tekton.py

```diff
@@ -494,24 +473,41 @@ def get_error_message(self):
         if not data:
             return None
 
-        annotations = data['metadata']['annotations']
+        task_runs_status = data['status'].get('taskRuns', {})
+        sorted_tasks = get_sorted_task_runs(task_runs_status)
 
-        plugins_metadata = annotations.get('plugins-metadata')
         plugin_errors = None
+        annotations_str = None
+
+        for _, stats in sorted_tasks:
+            if annotations_str is not None:
+                break
+
+            task_name = stats['pipelineTaskName']
+
+            if task_name not in ('binary-container-exit', 'source-container-exit'):
+                continue
+
+            for result in stats['status']['taskResults']:
+                if result['name'] != 'annotations':
+                    continue
+
+                annotations_str = result['value']
+                break
```

Comment:
We have very similar code in get_final_platforms. Could we generalize it? E.g. a `get_task_results` method which returns every result from every taskRun?
https://github.com/containerbuildsystem/osbs-client/blob/e53b2cc3923c7013bc84c77de255bc568c4968a9/osbs/tekton.py#L542

Reply:
done